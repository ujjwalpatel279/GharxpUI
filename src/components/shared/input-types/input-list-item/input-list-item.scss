// ========================================================================
// COMPONENTS/SHARED/INPUT-TYPES/INPUT-LIST-ITEM
// ========================================================================

@import '~sassHelpers';

.c-input-list-item {
  position: relative;
  display: inline-block;
  overflow: hidden;
  cursor: pointer;

  &__inner {
    display: flex;
    align-items: center;
    justify-content: center;
    height: spacer(8);
    padding-left: spacer(2);
    padding-right: spacer(2);
    background-color: white;
    border: 1px solid color-brand(secondary);
    border-radius: $border-radius;
  }

  input {
    position: absolute;
    top: -100px;
    left: -100px;

    &:focus ~ .c-radio-input__inner {
      outline: 1px dashed color-brand(secondary);
      outline-offset: -4px;
    }

    &:checked ~ .c-radio-input__inner {
      background-color: color-brand(secondary);
      color: white;
    }
  }
}
