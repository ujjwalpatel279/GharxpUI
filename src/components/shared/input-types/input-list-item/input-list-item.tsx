import React, { ChangeEvent, forwardRef } from 'react';

import './input-list-item.scss';

interface InputListItemProps {
  type: 'radio' | 'checkbox';
  id: string | number;
  name: string;
  value: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  disabled?: boolean;
}

type Ref = HTMLInputElement;

export const InputListItem = forwardRef<Ref, InputListItemProps>(
  ({ type, id, name, value, onChange, disabled }, ref) => (
    <label htmlFor={id.toString()} className="c-input-list-item">
      <input
        type={type}
        id={id.toString()}
        name={name}
        defaultValue={value}
        ref={ref}
        onChange={onChange}
        disabled={disabled}
      />
      <span className="c-radio-input__inner">
        <span>{value}</span>
      </span>
    </label>
  ),
);

InputListItem.displayName = 'InputListItem';
