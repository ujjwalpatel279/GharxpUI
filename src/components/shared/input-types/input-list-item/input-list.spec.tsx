import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { InputListItem } from './input-list-item';

describe('Input List Item Component', () => {
  afterEach(cleanup);

  it('renders the component', () => {
    const { container } = render(
      <InputListItem type="radio" name="inputName" value="inputValue" id="inputList" ref={() => null} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
