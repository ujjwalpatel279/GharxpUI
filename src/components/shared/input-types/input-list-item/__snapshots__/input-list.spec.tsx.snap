// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input List Item Component renders the component 1`] = `
<label
  class="c-input-list-item"
  for="inputList"
>
  <input
    id="inputList"
    name="inputName"
    type="radio"
    value="inputValue"
  />
  <span
    class="c-radio-input__inner"
  >
    <span>
      inputValue
    </span>
  </span>
</label>
`;
